tools/workflow/clean_aws_rsc.go

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package main
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"log"
+	"time"
+	"strings"
+)
+const (
+	containLbString = "aoc-lb"
+	pastDayDelete = 5
+)
+
+func main() {
+
+	log.Printf("Beging terminating EC2 Instances")
+	terminateEc2InstancesError := terminateEc2Instances()
+	if (terminateEc2InstancesError != nil){
+		log.Fatalf("Exit terminating EC2 Instances")
+	}
+
+	log.Printf("Begin destroy Load Balancer resources")
+	destroyLoadBalancerError := destroyLoadBalancerResource()
+
+	if (destroyLoadBalancerError != nil){
+		log.Fatalf("Exit destroy Load Balancer Resources")
+	}
+
+	log.Printf("Finish destroy AWS resources")
+}
+
+func terminateEc2Instances()(error){
+	// set up aws go sdk ec2 client
+	testSession := session.Must(
+		session.NewSessionWithOptions(
+			session.Options{
+				SharedConfigState: session.SharedConfigEnable,
+			}))
+
+	ec2client := ec2.New(testSession)
+
+	// Get list of instance
+	//State filter
+	instanceStateFilter := ec2.Filter{Name: aws.String("instance-state-name"), Values: []*string{aws.String("running")}}
+
+	//Tag filter
+	instanceTagFilter := ec2.Filter{Name: aws.String("tag:Name"), Values: []*string{aws.String("Integ-test-Sample-App"), aws.String("Integ-test-aoc")}}
+
+	//get instances to delete
+	var deleteInstanceIds []*string
+	var nextToken *string
+	for {
+		describeInstancesInput := ec2.DescribeInstancesInput{Filters: []*ec2.Filter{&instanceStateFilter, &instanceTagFilter}, NextToken: nextToken}
+		describeInstancesOutput, describeInstanceErr := ec2client.DescribeInstances(&describeInstancesInput)
+		if describeInstanceErr != nil {
+			log.Printf("Failed to get instance for error %v", describeInstanceErr)
+			return describeInstanceErr
+		}
+		for _, reservation := range describeInstancesOutput.Reservations {
+			for _, instance := range reservation.Instances {
+				//only delete instance if older than 5 days
+				if time.Now().UTC().Add(-1 * time.Hour * 24 * pastDayDelete).After(*instance.LaunchTime) {
```

Comment:
Can change the const to be the entire calculation instead of just the day or can have two consts: one with the day and another that performs this calculation. It's used here and in the load balancer deletion, so it might as well be reused.

Reply:
Reasonable. Updating with the latter solution (using two const).